CMakeLists.txt

```diff
@@ -92,11 +92,13 @@ if (UNIT_TESTING)
   # make generate step fail if cmocka dependency is not found
   find_package(cmocka CONFIG REQUIRED)
 
-  # When using VCPKG, cmocka lib name is set different than when it is globally installed
-  if(DEFINED ENV{VCPKG_ROOT} OR DEFINED ENV{VCPKG_INSTALLATION_ROOT})
-    set(CMOCKA_LIB ${CMOCKA_LIBRARIES})
-  else()
-    set(CMOCKA_LIB cmocka::cmocka)
+  # Patch for how cmocka was working before commit: https://github.com/microsoft/vcpkg/commit/fb8b20f4604eb495e1d772b16395980d171954ce
```

Comment:
We pin the commit/version of vcpkg to the this tag, which is **before** this commit. That's the version we support:
https://github.com/Azure/azure-sdk-for-c/blob/17ee1d158e4def2cb5a333a92a16b270847f52b3/eng/vcpkg-commit.txt#L1

So,
1) Do we still need this fix if the user is checking it out that version of vcpkg? If we follow the documented steps, do we still see the issue? https://github.com/Azure/azure-sdk-for-c#vcpkg
2) Would us upgrading to a newer release of vcpkg help solve or simplify the issue at all (2020.07 or possibly a newer one that might get released in the future)?

Reply:
1.- No, we don't need the fix if customers uses the previous commit of VCPKG. However, a customer might got latest vcpkg (like Cris did on Ubuntu 20) and face this issue. Then they would need to downgrade VCPKG to be able to build. That's not cool.
2.- It would be the same, IMO. If we move to latest, we still need to have this patch for old customer using an old VCPKG version.  It's not easy to ask a customer to upgrade VCPKG because that means re-building all installed packages, and it might be several apps.  (Maybe VCPKG has some upgrade option, I don't know)

Reply:
> However, a customer might got latest vcpkg (like Cris did on Ubuntu 20) and face this issue.

That isn't supported in our CI or docs. We expect users to follow the steps in the docs. Why would someone go get latest vcpkg? Since we don't test on those, that could have all sorts of behavioral breaks.